### ISSUE TITLE
Mapping high voltage (HV) lines in Pakistan, Nigeria and Zambia

### ISSUE THREAD

**Issue by dannykath** [developmentseed/mapping#14]
In collaboration with the World Bank, Development Seed is mapping HV electriticity infrastructure in Pakistan, Zambia and Nigeria, including the features `power=tower`, `power=line`, and `power=substation`. To assist us in this process, we are using a ML classifier to allow our team to do targeted mapping instead of searching through large areas for signs of infrastructure. The predictions of the model are only used as a contextual layer in JOSM , nothing is directly imported from the model into OSM. The tracing itself is being carried out by the Development Seed data team on the standard imagery layers available in OSM.

## Workflow and timeline

**Workflow**

1. Construct training/testing data sets from boundaries defined in geojsons 
2. Train Xception model over many sets of parameters
3. Evaluate model on test data using ROC analysis; choose best model
4. Download country-wide tile sets; use model to predict on every tile
5. Compile model predictions into a geojson map; A "positive" detection is only assigned to tiles where the score was above a threshold choosen from ROC analysis (here, 0.97)
6. Provide geojson map to the data team mappers to use as a guide
7. The data team adds visible HV towers, lines, and substations to OSM

**Summary of input/output for data team and model**

-  **Imagery for tracing**: standard layers available in OSM including Digital Globe, Bing, and Mapbox Satellite
-  **Output from the data team**: towers, powerlines, and substations mapped in OSM (see tagging guide below)
-  **ML model input**: RGB zoom 18 tiles from DG standard baselayer
-  **ML model output**: Prediction scores (on interval [0, 1]) indicating liklihood that a HV tower is present

**Timeline**

- **Jan 2018 (complete):**  Train, develop, and select model
- **Feb 2018 (in progress):** Use model to generate country-wide prediction maps
- **Feb 2018 (in progress):** Data team traces HV infrastructure and add it to OSM
- **Feb 2018:** Clean up code, publish approach, code, and model under an open source license 

## About the model

We use a deep learning framework (Tensorflow and Keras) to build a model that takes an RGB image and assigns the liklihood of a HV tower being present. Specifically, we use the Xception model architecture [Chollet, 2016](https://arxiv.org/abs/1610.02357) trained on three datasets where one set comes from each of the target countries. The training regions where these datasets originate will be released as part of the open source code (along with the saved model architecture and weights). The model predictions are compiled into a geojson map so that mappers can focus their energy on areas where there are strings of many positive predictions (as shown below). 

![image](https://user-images.githubusercontent.com/751330/35827579-893cd356-0a8a-11e8-8ef3-6d6abd2cdccb.png)
 **Purple**: existing OSM HV lines. **Orange**: tiles where the model predicts high probability of a tower.

#### [Tower](https://wiki.openstreetmap.org/wiki/Tag:power%3Dtower)
![power](https://user-images.githubusercontent.com/8483644/35824286-6b0d1832-0a80-11e8-8330-ea4f71d6b830.jpg)

#### [Line](https://wiki.openstreetmap.org/wiki/Tag:power%3Dline)
![lines](https://user-images.githubusercontent.com/8483644/35824442-e42b1016-0a80-11e8-8fe9-71b253cea9af.png)

#### [substation](https://wiki.openstreetmap.org/wiki/Tag:power%3Dsubstation) 
![station](https://user-images.githubusercontent.com/8483644/35824720-c68f0566-0a81-11e8-8d52-d6369ba6187d.jpg)

## Mapping sources for tracing

- Satellite Imagery:
  - DG Standard 
  - DG Premium
  - Bing

## Changeset Comment

 - Pakistan: `#mapping-hv-infrastructure #pakistan`
 - Nigeria: `#mapping-hv-infrastructure #nigeria`
 - Zambia: `#mapping-hv-infrastructure #zambia`

## Questions

If you have questions doubts about the mapping HV, satellite imagery or another, please post the question in https://github.com/developmentseed/mapping/issues/14

**Comment by olafveerman** [comment]
We completed mapping the grid in all three countries. A write-up of the methodology and results can be found here: http://devseed.com/ml-grid-docs/
